include/openssl/crypto.h.in

```diff
@@ -90,6 +90,60 @@ int CRYPTO_atomic_or(uint64_t *val, uint64_t op, uint64_t *ret,
                      CRYPTO_RWLOCK *lock);
 int CRYPTO_atomic_load(uint64_t *val, uint64_t *ret, CRYPTO_RWLOCK *lock);
 
+# if defined(OPENSSL_SYS_WINDOWS) || defined(_WIN32) || defined(__CYGWIN__)
+typedef long CRYPTO_SIGNAL_type;
+# else
+typedef int  CRYPTO_SIGNAL_type;
+# endif
+
+# define CRYPTO_SIGNAL_DEFAULT_ACTION NULL
+
+typedef void (*CRYPTO_SIGNAL_ACTION)(CRYPTO_SIGNAL_type);
+
+typedef struct {
+    long signal;
+    CRYPTO_SIGNAL_ACTION action;
+} CRYPTO_SIGNAL;
+
+int CRYPTO_SIGNAL_block(CRYPTO_SIGNAL* p);
+int CRYPTO_SIGNAL_block_set(CRYPTO_SIGNAL** props);
+int CRYPTO_SIGNAL_unblock_all(void);
+
+typedef void *CRYPTO_MUTEX;
```

Comment:
Or it could just be `typedef void CRYPTO_MUTEX;` in the same way that `CRYPTO_RWLOCK` is defined. Either way `*` should not be part of the type IMO.

Reply:
Does the mutex support even need to be public at this point?
Could it live in include/internal/thread_arch.h instead of include.openssl/crypto.h.in?

Reply:
Looking more deeply at this and the Argon PR, there doesn't seem to be a need to expose the mutex support beyond the threading code itself -- i.e. it's not necessary to make it available within other parts of libcrypto.

Given our very conservative policies about supporting pubilc APIs, I'd prefer for it to not be made public until it needs to be.  While it's internal, we can make changes.

Reply:
Thanks! Condvars and mutexes are now internal. CRYPTO_MUTEX is now typedef'd to void (previously void*).